omen2/table.py

```diff
@@ -206,34 +207,35 @@ def count(self, _where={}, **kws) -> int:
 
     @contextlib.contextmanager
     def _unsafe_transaction(self):
-        tid = threading.get_ident()
-        self._tx_objs[tid] = {}
-        needs_rollback: Set[Tuple["ObjBase", TxStatus]] = set()
-        try:
-            with self.db.transaction():
-                yield self
+        with ObjMeta.lock:
```

Comment:
i think this was not intended to be static.  i think it should probably be a per-object lock

we can fix this by:
- fix per object locks
- maintain set of active object locks in the table
- block new entries to the table and wait on all active locks before beginning a table-wide transaction

it's not a simple solution, but i think it's the correct one

Reply:
added per-object locks, and a table-level lock. please LMK what you think.

wondering if we should apply the VERY_LARGE_TIMEOUT across all of these to avoid deadlocks.

Reply:
makes sense, that sounds like it should be a separate change.